repose-aggregator/commons/utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -63,67 +76,275 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
+      setHeader(CommonHttpHeader.CONTENT_TYPE.toString, contentType)
+    }
+  }
 
-  override def getSplittableHeaders(headerName: String): util.List[String] = ???
+  override def getCharacterEncoding: String = {
+    Option(getHeader(CommonHttpHeader.CONTENT_TYPE.toString))
+      .map(_.split("charset=")(1))
+      .getOrElse("ISO-8859-1")
+  }
+
+  override def setCharacterEncoding(s: String): Unit = {
+    if (!isCommitted && responseBodyType != ResponseBodyType.PrintWriter) {
+      Option(getHeader(CommonHttpHeader.CONTENT_TYPE.toString))
+        .map(_.split(';')(0))
+        .foreach(contentType => setHeader(CommonHttpHeader.CONTENT_TYPE.toString, contentType + ";charset=" + s))
+    }
+  }
 
   /**
    * @throws IllegalStateException when bodyMode is ResponseMode.PASSTHROUGH
    */
-  def getOutputStreamAsInputStream(): InputStream = ???
+  def getOutputStreamAsInputStream: InputStream = bodyOutputStream.getOutputStreamAsInputStream
 
   /**
    * @throws IllegalStateException when bodyMode is anything other than ResponseMode.MUTABLE
    */
-  def setOutput(inputStream: InputStream): Unit = ???
+  def setOutput(inputStream: InputStream): Unit = bodyOutputStream.setOutput(inputStream)
+
+  override def getWriter: PrintWriter = {
+    responseBodyType match {
+      case ResponseBodyType.OutputStream =>
+        throw new IllegalStateException("Cannot call getWriter after calling getOutputStream")
+      case _ =>
+        responseBodyType = ResponseBodyType.PrintWriter
+        bodyPrintWriter
+    }
+  }
+
+  override def getOutputStream: ServletOutputStream = {
+    responseBodyType match {
+      case ResponseBodyType.PrintWriter =>
+        throw new IllegalStateException("Cannot call getOutputStream after calling getWriter")
+      case _ =>
+        responseBodyType = ResponseBodyType.OutputStream
+        bodyOutputStream
+    }
+  }
+
+  override def flushBuffer(): Unit = {
+    responseBodyType match {
+      case ResponseBodyType.PrintWriter => bodyPrintWriter.flush()
+      case _ => bodyOutputStream.flush()
+    }
+  }
+
+  override def resetBuffer(): Unit = {
+    if (isCommitted) {
+      throw new IllegalStateException("Cannot call resetBuffer after the response has been committed")
+    } else {
+      super.resetBuffer()
+      bodyOutputStream.resetBuffer()
+    }
+  }
+
+  override def reset(): Unit = {
+    if (isCommitted) {
+      throw new IllegalStateException("Cannot call reset after the response has been committed")
+    } else {
+      super.reset()
+      headerMap = new TreeMap[String, Seq[String]]()(caseInsensitiveOrdering)
+      resetBuffer()
+    }
+  }
 
   /**
-   * @throws IllegalStateException when neither headerMode nor bodyMode are ResponseMode.MUTABLE
+   * @throws IllegalStateException when neither headerMode nor bodyMode is ResponseMode.MUTABLE
    */
-  def commitToResponse(): Unit = ???
+  def commitToResponse(): Unit = {
```

Comment:
I thought we were going to have this commit the underlying if they had commited this one?